R/read_supp_cy.R

```diff
@@ -0,0 +1,76 @@
+#' Download, tidy and import ABS International Trade Supplementary Information by Calendar Year
+#'
+#' Downloads Tables 1 through 8, inclusive, of the ABS International Trade: Supplementary Information data by Calendar Year.
+#' @param table_no Selects which tables from the ABS will be imported
+#' @param path Path where Excel files downloaded from the ABS should be stored
+#' @return A tibble containing the relevant table, or a list of tibbles if more than one ABS table has been requested
+#' @examples
+#' \dontrun{
+#' read_supp_cy()
+#' }
+#' @export
+
+read_supp_cy <- function(table_no = c(1, 2, 3, 4, 5, 6, 7, 8), path = tempdir()) {
+  temp <- tempfile()
+
+  c_year <- rvest::read_html("https://www.abs.gov.au/statistics/economy/international-trade/international-trade-supplementary-information-calendar-year/latest-release")
+
+  c_year_url <- c_year %>%
+    rvest::html_nodes("a") %>%
+    rvest::html_attr("href") %>%
+    stringr::str_subset(".zip")
+
+  download.file(c_year_url, temp)
+
+  unzip(temp, exdir = path)
+
+  unlink(temp)
+
+  file_no <- length(list.files(path = path, pattern = "*.xls"))
+
+  if (file_no >= 15) {
```

Comment:
Can you talk me through this line, @Louis-DJPR?

Reply:
Hi Matt, I have used the number of .xls files in a directory to equal the number of tables that the script will generate. I noticed that locally the .xls files aren't removed from the directory after the script finishes so if you run the script for both calendar year and financial year data it will throw out an error (as there are twice as many excel files as the script expects) as I result I just did a bit of a silly fix and halved the number if R detects too many excel files.

Reply:
New code does not require this trick to be used as it clears `tempdir()` in the function.